src/test/java/net/minestom/server/entity/player/PlayerRespawnChunkTest.java

```diff
@@ -0,0 +1,103 @@
+package net.minestom.server.entity.player;
+
+import net.minestom.server.MinecraftServer;
+import net.minestom.server.coordinate.Pos;
+import net.minestom.server.entity.Player;
+import net.minestom.server.event.player.PlayerChunkLoadEvent;
+import net.minestom.server.event.player.PlayerChunkUnloadEvent;
+import net.minestom.server.network.packet.client.play.ClientStatusPacket;
+import net.minestom.server.network.packet.server.play.ChunkDataPacket;
+import net.minestom.server.network.packet.server.play.UnloadChunkPacket;
+import net.minestom.server.utils.chunk.ChunkUtils;
+import net.minestom.testing.Env;
+import net.minestom.testing.EnvTest;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+
+@EnvTest
+public class PlayerRespawnChunkTest {
+
+
+    @Test
+    public void testChunkUnloadsOnRespawn(Env env) {
+        var instance = env.createFlatInstance();
+        var connection = env.createConnection();
+        Player player = connection.connect(instance, new Pos(0, 40, 0)).join();
+        player.teleport(new Pos(32, 40, 32)).join();
+
+        var unloadChunkTracker = connection.trackIncoming(UnloadChunkPacket.class);
+        player.setHealth(0);
+        player.respawn();
+        // Since client unloads the chunks, we shouldn't receive any unload packets
+        unloadChunkTracker.assertCount(0);
+    }
+
+    @Test
+    public void testChunkReloadCount(Env env) {
+        var instance = env.createFlatInstance();
+        var connection = env.createConnection();
+        Player player = connection.connect(instance, new Pos(0, 40, 0)).join();
+
+        var loadChunkTracker = connection.trackIncoming(ChunkDataPacket.class);
+        player.setHealth(0);
+        player.respawn();
+        // Player should have all their chunks reloaded
+        int chunkLoads = ChunkUtils.getChunkCount(MinecraftServer.getChunkViewDistance());
+        loadChunkTracker.assertCount(chunkLoads);
+    }
+
+    @Test
+    public void testPlayerTryRespawn(Env env) {
+        var instance = env.createFlatInstance();
+        var connection = env.createConnection();
+        Player player = connection.connect(instance, new Pos(0, 40, 0)).join();
+
+        player.eventNode().addListener(PlayerChunkLoadEvent.class, playerChunkLoadEvent -> {
```

Comment:
There is no reason to load any chunk when the player respawn at the same position. Packets being sent is an implementation detail